### ISSUE TITLE
[feature request] seqkit subseq with --circular option?

### ISSUE THREAD

**Issue by xinehc** [shenwei356/seqkit#475]
Hi Wei,

I wonder would it be possible to add a `--circular` option also to `seqkit subset`?

I am currently using `seqkit subset` to extract the flanking regions of some genes with bed files. Some genes are located near the boundary of circular genomes so the flanks get truncated. It would be nice to have this option so that I don't need to manually concatenate the sequences multiple times before annotating genes. It would be even nicer if `seqkit subset` could differentiate linear/circular genomes with a user specified list or bed. For example:

```bash
seq_a    linear
seq_b    circular
...
```

**Comment by shenwei356** [comment]
seqkit does not have a `subset` command.

> seqkit subseq with --circular option

It sounds useful.

What's the organism the sequence belong to? Linear human and circular mitochondria?  Adding a new column would break the format specification of BED/GTF. You can simply run with linear and circular sequence separately.

**Comment by xinehc** [comment]
Thanks for the quick reply, I meant `seqkit grep`, sorry for the confusion.

I am working with a set of plasmid sequences which are short and circular in general, but there might be cases that the plasmid is incomplete and thereby linear.

**Comment by shenwei356** [comment]
`grep` does not handle flanking sequence. subseq?

**Comment by xinehc** [comment]
You are right. I used both `subseq` and `grep` in a project and got confused.
